research_amp/sports_prediction/sota_baseline.py

```
@@ -0,0 +1,319 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.15.2
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %%
+import sqlite3
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import itertools
+import matplotlib.pyplot as plt
+import os
+import calendar
+from datetime import datetime, timedelta, date
+import zipfile
+import re
+from sklearn.model_selection import train_test_split, GridSearchCV
+from scipy.stats import gaussian_kde
+from sklearn.model_selection import StratifiedShuffleSplit
+import matplotlib.pyplot as plt
+from scipy.stats import kde
+from sklearn.preprocessing import StandardScaler
+from sklearn.impute import SimpleImputer, KNNImputer
+import boto3
+from io import BytesIO
+from scipy.optimize import minimize
+import warnings
+from joblib import Parallel, delayed
+import math
+pd.set_option('display.max_columns', None)
+warnings.filterwarnings("ignore")
+
+# %% [markdown]
+# Set the s3 credentials
+
+# %%
+# Initialize a session using Amazon S3 with a specific profile
+session = boto3.Session(profile_name='ck')
+s3 = session.client('s3')
+
+# Set the S3 bucket and dataset path
+s3_bucket_name = 'cryptokaizen-data-test'
+s3_dataset_path = 'kaizen-ai/datasets/football/OSF_football/'
+
+# Define the local directory to save the files
+local_directory = 'datasets/football/OSF_football'
+os.makedirs(local_directory, exist_ok=True)
+
+# Function to download files from S3
+def download_files_from_s3(bucket, prefix, local_dir):
+    try:
```

why try-catch?
Let it fail